### PR TITLE
Delete terminated sessions in batches

### PR DESCRIPTION
**Pull request by mgaffney**
This PR:

* adds retries to `scheduler.JobStatus`
* adds a new `batch` package under the `scheduler` package
* uses the new `batch` package to delete terminated sessions in batches

### CODE REVIEW

**Review by tmessi — CHANGES_REQUESTED** [review]

**Review by johanbrandhorst — APPROVED** [review]
LGTM!

File: internal/scheduler/job.go [hunk 1]
```diff
@@ -11,20 +11,32 @@ import (
...
-	// Status reports the job’s current status.  The status is periodically persisted by
-	// the scheduler when a job is running, and will be used to verify a job is making progress.
+	// Status reports the job’s current status. It is called periodically
+	// while the Run method is running and immediately after the Run method
+	// completes.
+	//
+	// The scheduler uses the values in the returned JobStatus to verify
+	// that progress is being made by the job. The scheduler will interrupt
+	// the job if the values returned by Status do not change over a
+	// configurable amount of time.
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/scheduler/job.go:23]: Do you mean the `Scheduler` structs `interruptThreshold` field?

File: internal/scheduler/batch/batch.go [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+
+	// StatusThreshold is the amount of time the job has to return a
+	// JobStatus with values different from the previous JobStatus. If a
+	// call to Exec approaches this threshold, Batch will interrupt the
+	// call to Exec, reduce the batch size, and call Exec again. It will
+	// also increment the Retries value reported in the Status method.
+	//
+	// If StatusThreshold is less than or equal to zero, it will be set to
+	// DefaultStatusThreshold.
+	StatusThreshold time.Duration
```
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/scheduler/batch/batch.go:107]: The StatusThreshold is actually set by the job scheduler. I'm only trying to cover the zero value and the odd case that someone puts in a negative number.

File: internal/scheduler/job.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mgaffney) [reply internal/scheduler/job.go:23]: I updated the comment

**Review by tmessi — APPROVED** [review]

**Pull request merged**